src/RemoteAPI/related/class-related-caching-decorator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Parsely Related REST API Caching Decorator
+ *
+ * @package Parsely
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\RemoteAPI;
+
+use WP_Object_Cache;
+
+/**
+ * Caching Decorator for the remote /related endpoint.
+ */
+class Related_Caching_Decorator implements Proxy {
+	/**
+	 * The Proxy instance this will cache.
+	 *
+	 * @var Proxy
+	 */
+	private $decorated_proxy;
+
+	/**
+	 * Usually $GLOBALS['wp_object_cache'], but automated tests can inject a mock.
+	 *
+	 * @var WP_Object_Cache
+	 */
+	private $cache;
+
+	/**
+	 * The computed cache key.
+	 *
+	 * @var string
+	 */
+	private $cache_key;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Proxy           $proxy The Proxy object to cache.
+	 * @param WP_Object_Cache $cache The WordPress object cache instance.
+	 */
+	public function __construct( Proxy $proxy, WP_Object_Cache $cache ) {
+		$this->decorated_proxy = $proxy;
+		$this->cache           = $cache;
+		$this->cache_key       = 'api_recos-' . wp_hash( wp_json_encode( $this->decorated_proxy->get_query() ) );
```

Comment:
Does this cache key prefix need updating to reflect we're now using the nomenclature `related`?